### DISCUSSION TITLE
Initial conditions questions - (doesn't take second order data, simulation speed)

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#17314] [Q&A General] [answered]
Hi all,

I have 2 questions regarding MOOSE initial conditions

First question is does 'initial conditions' work as the initial guess forthe solver?  if so, I don't really see much of an imporvement at all, maybe I did something wrong?

Here is a rough workflow of my simulation: solve a steady state problem > upload file to the next simulation > change the material property (young's modulus/density/possion) of some block IDs > solve a steady state problem > repeat.

Second question: I tried to load the 2nd order Auxvariable from .e file
```

[Mesh]
[./gen]
   type = FileMeshGenerator
     file = gravz_later_step1_Pre_backfilltest.e
     show_info = true
     use_for_exodus_restart = true
     # parallel_type = distributed
  []
[]

[AuxVariables]
  [./s11_aux]
    order = SECOND
    family = MONOMIAL
   initial_from_file_var = s11_aux
  [../]
  [./s22_aux]
    order = SECOND
    family = MONOMIAL
   initial_from_file_var = s22_aux
  [../]
  [./s33_aux]
    order = SECOND
    family = MONOMIAL
   initial_from_file_var = s33_aux
  [../]
[]
```
and I received this error
`Error! Trying to copy elemental solution into a variable that is not of CONSTANT MONOMIAL type.`

It has no error taking 2nd order variable
```
[Variables]
    [./disp_x]
      order = SECOND
      family = LAGRANGE
       initial_from_file_var = disp_x
    [../]
    [./disp_y]
      order = SECOND
      family = LAGRANGE
      initial_from_file_var = disp_y
    [../]
    [./disp_z]
      order = SECOND
      family = LAGRANGE
       initial_from_file_var = disp_z
    [../]
[]
```
is there any way to work around this? can I calculate Auxvariable pre and post simulation?

Thank you guys!

Regards,
Traiwit

**Comment by ngrilli** [comment, marked as answer]
Dear @Traiwit 

1) For time dependent problems, the initial conditions should be the initial guess of the solver.
Not sure about steady state problems. Initial conditions may not have any effect in that case.

2) Not sure if that is possible. Were the AuxVariables in the imported file also second order monomial?

Nicolò

**Reply by WilkAndy** [reply]
Hang on, although i'm sure you know about initial conditions, @ngrilli , i think your wording is confusing, so for posterity i'll write:
- For Variables, initial conditions are the physical initial conditions of your system.   They are usually well-known.  For instance, displacement = 0, or pressure = atmospheric, or temperature = room_temperature, etc.  For `steady` simulations, the initial conditions don't matter *in principal*, because MOOSE will find the solution no-matter what the initial conditions you set (assuming there is a unique solution) but *in practice* initial conditions are extremely important, since if your ICs are close to the final steady-state solution MOOSE will find it easy to converge, while if you just set random ICs MOOSE will take ages to converge.  In this sense, for `steady` problems, the ICs can be thought of as "initial guess".
- For AuxVariables, initial conditions can be more complicated.  Firstly, they could be over-written by your AuxKernels so the `initial` may have no impact.  Secondly, some AuxVariables are simply stores of information and aren't actually used in Kernels/Materials/etc, so have no impact on the solution process, and setting an `initial` is just for visualisation, etc.  On the other hand, some AuxVariables feed into Kernels/etc, so their `initial` may be very important, and you should choose something physical (for time-dep problems) or reasonable (for steady problems).

**Reply by Traiwit** [reply]
@WilkAndy may I ask a really dumb question,

In Tensormechanic, seem like the only variable that goes into the solver is 'disp_', would importing the variable from a file as follow (below) enough to let MOOSE know that I want to use these disp_ as ICs?

Thank you!

```
[Variables]
    [./disp_x]
      order = SECOND
      family = LAGRANGE
       initial_from_file_var = disp_x
    [../]
    [./disp_y]
      order = SECOND
      family = LAGRANGE
      initial_from_file_var = disp_y
    [../]
    [./disp_z]
      order = SECOND
      family = LAGRANGE
       initial_from_file_var = disp_z
    [../]
[]
```

**Reply by WilkAndy** [reply]
Yes, that is the syntax you should use to set your displacement ICs.

**Reply by Traiwit** [reply]
Thank you @WilkAndy ,

For the 2nd question: Yes @ngrilli, the input file contains the second order monomial data.

**Reply by WilkAndy** [reply]
Hey @traiwit, i just want to highlight the phrase "assuming there is a unique solution" in my post above.  Just remember that when you start using plasticity, your solution will be path dependent, eg, will depend on your excavation schedule (and any damping parameters you include).  Eg, if you excavate the entire mine in one big chunk, the overburden will tend to fall down in one big chunk (with failure only above the boundary of your excavation), whereas if you excavate in many small steps, the rock will tend to fail and bulk-up throughout the entire overburden.

**Reply by Traiwit** [reply]
@WilkAndy thanks I will keep that in mind, i'm still not happy with my elastic model, might be some time until I can move to plastic model.

I've been reading the coal-mine example too, I hope everything makes sense eventually :)
